src/descriptor/key.rs

```diff
@@ -18,15 +18,19 @@ use crate::prelude::*;
 use crate::serde::{Deserialize, Deserializer, Serialize, Serializer};
 use crate::{hash256, MiniscriptKey, ToPublicKey};
 
+type DescriptorSinglePublicKey = SinglePub;
+type DescriptorExtendedPublicKey = DescriptorXKey<bip32::ExtendedPubKey>;
+type DescriptorMultiExtendedPublicKey = DescriptorMultiXKey<bip32::ExtendedPubKey>;
```

Comment:
IMO we should not start putting `Extended` back into the identifiers of keys. The only other place that appears is in `bitcoin::bip32` which is not a module to copy :)